src/pip/_internal/req/req_install.py

```diff
@@ -777,18 +777,47 @@ def install(
 
         global_options = global_options if global_options is not None else []
         if self.editable:
-            install_editable_legacy(
-                install_options,
-                global_options,
-                prefix=prefix,
-                home=home,
-                use_user_site=use_user_site,
-                name=self.name,
-                setup_py_path=self.setup_py_path,
-                isolated=self.isolated,
-                build_env=self.build_env,
-                unpacked_source_directory=self.unpacked_source_directory,
-            )
+            if self.metadata_directory and self.metadata_directory.endswith('.dist-info'):
+
+                editable_path = self.pep517_backend.build_editable()
+
+                # create empty RECORD
+                with open(os.path.join(self.metadata_directory, 'RECORD'), 'w+'):
+                    pass
+
+                wheeldir = os.path.dirname(self.metadata_directory)
+
+                # put .pth file in wheeldir
+                with open(os.path.join(wheeldir, self.name + '.pth'), 'w+') as pthfile:
+                    pthfile.write(editable_path['src_root'] + '\n')
```

Comment:
@dholth @pfmoore Commenting here because it might be easier than on the discuss thread.

So why not let the back-end put the .pth file in the wheel?
For simple cases the back-end just adds a .pth file which points to the source directory.
For advanced cases the back-end can also add an import hook to remap or filter the source.

The front-end does not need to know exactly *how* the back-end does it, it just needs to install a wheel.

Reply:
@sbidoul That's pretty much the discussion going on in Discourse right now 🙂 We have the start of a prototype that basically allows the backend to provide a completely standard wheel that implements an editable install.

So all the frontend needs to do is send a flag on `build_wheel` to tell the backend to make an editable wheel. (Either via `config_settings` as a prototype, or as a new argument in a final version). Everything else works unchanged.

Reply:
Ok perfect then. That's what I had understood yesterday evening, but I was confused with the newest messages this morning.

Reply:
Yep, things are moving fairly fast as people (or at least, me!) discover how much power Python's import mechanisms have 🙂 

One proviso - this isn't going to work on Python 2, so a wheel that just installs a `.pth` file might be the limit there. Or we could wait a few months until we drop support for Python 2 😄

Reply:
> One proviso - this isn't going to work on Python 2, so a wheel that just installs a .pth file might be the limit there

In the end it will depend if the backend supports python 2 or not. For pip there will be no difference.

Reply:
I let the backend provide the wheel. n.b. discourse